tests/unit/s2n_config_test.c

```diff
@@ -731,5 +731,204 @@ int main(int argc, char **argv)
+         * wasn't added until OpenSSL 1.1.0.
+         */
+#if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0)
+        {
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new_minimal(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+            EXPECT_NULL(config->trust_store.trust_store);
+            EXPECT_FALSE(config->trust_store.loaded_system_certs);
+
+            /* Initialize the X509_STORE by adding a cert */
+            EXPECT_SUCCESS(s2n_x509_trust_store_from_ca_file(&config->trust_store, S2N_RSA_PSS_2048_SHA256_CA_CERT, NULL));
+            EXPECT_NOT_NULL(config->trust_store.trust_store);
+            EXPECT_FALSE(config->trust_store.loaded_system_certs);
+
+            /* The X509_STORE should only contain the single cert that was added. */
+            STACK_OF(X509_OBJECT) *x509_store_contents = X509_STORE_get0_objects(config->trust_store.trust_store);
+            EXPECT_NOT_NULL(x509_store_contents);
+            int initial_contents_count = sk_X509_OBJECT_num(x509_store_contents);
+            EXPECT_EQUAL(initial_contents_count, 1);
+
+            /* Override the system cert file to guarantee that a system cert will be loaded */
+            EXPECT_SUCCESS(setenv("SSL_CERT_FILE", S2N_SHA1_ROOT_SIGNATURE_CA_CERT, 1));
+
+            /* Load the system cert into the store */
+            EXPECT_SUCCESS(s2n_config_load_system_certs(config));
+            EXPECT_TRUE(config->trust_store.loaded_system_certs);
+            int system_certs_contents_count = sk_X509_OBJECT_num(x509_store_contents);
+
+            /* LibreSSL doesn't use the SSL_CERT_FILE environment variable to set the system cert location,
+             * so we don't know how many system certs will be loaded, if any.
+             */
+            if (!s2n_libcrypto_is_libressl()) {
+                EXPECT_EQUAL(system_certs_contents_count, initial_contents_count + 1);
+            }
+
+            /* Additional calls to s2n_config_load_default_certs should not add additional certs to the store */
+            for (int i = 0; i < 20; i++) {
+                EXPECT_FAILURE_WITH_ERRNO(s2n_config_load_system_certs(config), S2N_ERR_X509_TRUST_STORE);
+                EXPECT_TRUE(config->trust_store.loaded_system_certs);
+                int additional_call_contents_count = sk_X509_OBJECT_num(x509_store_contents);
+                EXPECT_TRUE(additional_call_contents_count == system_certs_contents_count);
+            }
+
+            EXPECT_SUCCESS(unsetenv("SSL_CERT_FILE"));
+        }
+#endif
+
+        /* Self-talk tests */
+        {
+            DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key = NULL,
+                    s2n_cert_chain_and_key_ptr_free);
+            EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                    S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+            /* Ensure a handshake succeeds with a minimal server config and no mutual auth */
```

Comment:
I'm not sure this test is necessary-- isn't it just a very basic handshake, with no special trust store related behavior?

Reply:
Yeah, that's true. Although, it might be good to have at least one self-talk test that succeeds when using s2n_config_new_minimal, since all other tests are using s2n_config_new. But yeah, maybe it's unnecessary.